src/main.cc

```diff
@@ -0,0 +1,79 @@
+#include <bits/stdc++.h>
+#ifdef ONPC
+#include "../algo/debug.h"
+#else
+#define debug(...) 42
+#endif
+#define fr first
+#define sc second
+#define pb push_back
+using namespace std;
+
+typedef long long ll;
+typedef long double ld;
+typedef unsigned long long ull;
+
+template<typename T, typename Functor>
+struct MinimaQueue {
```

Comment:
No longer "Minima" i think, make FunctorQueue and remove min as default functor.